drivers/gnss/quectel_lcx6g.c

```diff
@@ -0,0 +1,368 @@
+	static struct modem_chat_script _sym = {                                                \
+		.name = #_sym,                                                                  \
+		.script_chats = _script_chats,                                                  \
+		.script_chats_size = ARRAY_SIZE(_script_chats),                                 \
+		.abort_matches = NULL,                                                          \
+		.abort_matches_size = 0,                                                        \
+		.callback = _callback,                                                          \
+		.timeout = _timeout,                                                            \
+	}
+
+#ifdef CONFIG_PM_DEVICE
+MODEM_CHAT_MATCH_DEFINE(pair003_success_match, "$PAIR001,003,0*38", "", NULL);
+MODEM_CHAT_SCRIPT_CMDS_DEFINE(
+	suspend_script_cmds,
+	MODEM_CHAT_SCRIPT_CMD_RESP("$PAIR003*39", pair003_success_match)
+);
+
+MODEM_CHAT_SCRIPT_NO_ABORT_DEFINE(suspend_script, suspend_script_cmds,
+				  quectel_lcx6g_script_cb, QUECTEL_LCX6G_SCRIPT_TIMEOUT_S);
+#endif /* CONFIG_PM_DEVICE */
+
+MODEM_CHAT_MATCH_DEFINE(pair002_success_match, "$PAIR001,002,0*39", "", NULL);
+MODEM_CHAT_MATCH_DEFINE(pair062_ack_match, "$PAIR001,062,0*3F", "", NULL);
+char ppp_setup_script_cmd_buf[sizeof("$PAIR752,x,xxx*xx")];
+MODEM_CHAT_MATCH_DEFINE(pair752_success_match, "$PAIR001,752,0*3B", "", NULL);
+MODEM_CHAT_SCRIPT_CMDS_DEFINE(
+	resume_script_cmds,
+	MODEM_CHAT_SCRIPT_CMD_RESP("$PAIR002*38", pair002_success_match),
+	MODEM_CHAT_SCRIPT_CMD_RESP("$PAIR062,0,1*3F", pair062_ack_match),
+	MODEM_CHAT_SCRIPT_CMD_RESP("$PAIR062,1,0*3F", pair062_ack_match),
+	MODEM_CHAT_SCRIPT_CMD_RESP("$PAIR062,2,0*3C", pair062_ack_match),
+	MODEM_CHAT_SCRIPT_CMD_RESP("$PAIR062,3,5*38", pair062_ack_match),
+	MODEM_CHAT_SCRIPT_CMD_RESP("$PAIR062,4,1*3B", pair062_ack_match),
+	MODEM_CHAT_SCRIPT_CMD_RESP("$PAIR062,5,0*3B", pair062_ack_match),
+	MODEM_CHAT_SCRIPT_CMD_RESP(ppp_setup_script_cmd_buf, pair752_success_match),
+);
+
+MODEM_CHAT_SCRIPT_NO_ABORT_DEFINE(resume_script, resume_script_cmds,
+				  quectel_lcx6g_script_cb, QUECTEL_LCX6G_SCRIPT_TIMEOUT_S);
+
+static void quectel_lcx6g_gga_cb(struct modem_chat *chat, char **argv, uint16_t argc,
+				 void *user_data)
+{
+	struct quectel_lcx6g_data *data = (struct quectel_lcx6g_data *)user_data;
+
+	if (gnss_nmea0183_parse_gga((const char **)argv, argc, &data->data) < 0) {
+		return;
+	}
+}
+
+static void quectel_lcx6g_rmc_cb(struct modem_chat *chat, char **argv, uint16_t argc,
+				 void *user_data)
+{
+	struct quectel_lcx6g_data *data = (struct quectel_lcx6g_data *)user_data;
+	int ret;
+
+	ret = gnss_nmea0183_parse_rmc((const char **)argv, argc, &data->data);
+	if (ret < 0) {
+		return;
+	}
+
+	gnss_publish_data(data->dev, &data->data);
+}
+
+MODEM_CHAT_MATCHES_DEFINE(unsol_matches,
+	MODEM_CHAT_MATCH_WILDCARD("$??GGA,", ",*", quectel_lcx6g_gga_cb),
+	MODEM_CHAT_MATCH_WILDCARD("$??RMC,", ",*", quectel_lcx6g_rmc_cb),
+);
```

Comment:
oh that's interesting, did not realize you would have ended up using the modem chat stuff like this (bet you mentioned it in the proposal a million times and I didn't connect). :-) So this should be usable for a generic "dumb" GPS driver as well right? One that just powers on and starts printing NMEA sentences, no need for the gnss driver to handle the UART config manually.

Reply:
That was the plan https://www.youtube.com/watch?v=QyrDgEz3DR0 :)

The GNSS driver is essentially a "dumb" driver ATM, it doesn't use the UART directly either, it uses the modem_uart_backend that can be exchanged for any other backend based on the bus used :)